Source/pyboy/pyboy.py

```diff
@@ -124,50 +207,213 @@ def stop(self, save=True):
 
-    def saveState(self, filename):
-        self.mb.save_state(filename)
+    def get_screen_position(self):
+        """
+        These coordinates define the offset in the tile map from where the top-left corner of the screen is place. Note that the tile map defines 256x256 pixels, but the screen can only show 160x144 pixels. When the offset is closer to the right or bottom edge than 160x144 pixels, the screen will wrap around and render from the opposite site of the tile map.
 
-    def loadState(self, filename):
-        self.mb.load_state(filename)
+        For more details, see "7.4 Viewport" in the [report](https://github.com/Baekalfen/PyBoy/raw/master/PyBoy.pdf), or the Pan Docs under [LCD Position and Scrolling](http://bgb.bircd.org/pandocs.htm#lcdpositionandscrolling).
 
-    def getSerial(self):
+        Returns:
+            ((int, int), (int, int)): Returns the registers (SCX, SCY), (WX - 7, WY)
+        """
+        return (self.mb.lcd.getviewport(), self.mb.lcd.getwindowpos())
+
+    def save_state(self, file_handle):
+        """
+        Saves the complete state of the emulator. It can be called at any time, and enable you to revert any progress in a game.
+
+        You can either save it to a file, or in-memory. The following two examples will provide the file handle in each case. Remember to `seek` the in-memory buffer to the beginning before calling `PyBoy.load_state`:
+
+            # Save to file
+            file_handle = open("state_file.state", "wb")
+
+            # Save to memory
+            import io
+            file_handle = io.BytesIO()
+            file_handle.seek(0)
+
+        Args:
+            file_handle (io.BufferedIOBase): A file-like object for which to write the emulator state.
+        """
+        self.mb.save_state(file_handle)
+
+    def load_state(self, file_like_object):
+        """
+        Restores the complete state of the emulator. It can be called at any time, and enable you to revert any progress in a game.
+
+        You can either load it from a file, or from memory. See `PyBoy.save_state` for how to save the state, before you can load it here.
+
+        Args:
+            file_handle (io.BufferedIOBase): A file-like object for which to read the emulator state.
+        """
+        self.mb.load_state(file_like_object)
+
+    def get_serial(self):
+        """
+        Provides all data that has been sent over the serial port since last call to this function.
+
+        Returns:
+            str : Buffer data
+        """
         return self.mb.getserial()
 
-    def disableTitle(self):
+    def disable_title(self):
+        """
+        Disable window title updates. These are output to the log, when in `headless` or `dummy` mode.
+        """
         self.window.disable_title()
 
-    def setEmulationSpeed(self, v, max_speed=0):
-        self.set_emulation_speed(v, max_speed)
+    def set_emulation_speed(self, limit, target_speed=0):
```

Comment:
This description doesn't seem to match the actual behavior in the code. If `self.limit_emulationspeed` is true, then `window.frame_limiter(speed)` is called with `speed=1`, and a false value will result in `frame_limiter` being called with the `max_emulationspeed` value. Perhaps line 413 should read "If *not* limiting the speed,..." ?

Reply:
I've now changed it to a single number. It's the target speed, or 0 for unlimited speed.

Reply:
I had the idea, that the user might want an option to switch between 1x and 5x speed, when the game executes way too fast in some games at unlimited speed.